oauth.rst

```diff
@@ -1,131 +1,129 @@
 
-- App name
-- App logo
-- App description
+Detailed information
+--------------------
+To make sense of the above it can be useful to observe the whole flow in a detailed step-by-step explanation. The following steps explain how the flow would look like for a Client getting an Access Token to act on behalf of one of its users, and then using it to get user data from Dock:
 
-And also some URLs:
+Step 1 - Authorization grant query
+----------------------------------
+To initiate the OAuth flow the Client needs to start an authorization grant query by redirecting the user to:
+``GET https://app.dock.io/oauth/authorize`` with the following query params:
 
-- Homepage url
-- Privacy policy url
-- Callback URIs (this is a list)
+- ``client_id``: id of the Client doing the request. This is the id you received when you registered your application.
+- ``response_type=code``: 'code' is always expected, it means that you expect to get an ``authorization_code``
+- ``redirect_uri``: URL where you want the user to be redirected to by the Authorization Server if the authorization grant query is successful. This needs to be one of the Redirect URIs you added when registering your application. Remember that **exact match** is used to compare, any minor variations will trigger an error response.
+- ``scope``: scope name(s). These define which of the user's resources you're requesting access to. Our API expects comma-separated values if more than one.
+- ``state``: (optional but recommended) the value of this parameter will be returned to you unmodified. We recommend that you encode some info and sign it. It is useful for you to verify the authenticity of the next call you receive from us, if the signature doesn't match then you can safely abort the flow.
 
-The given information, except for the callback URIs, is shown to the User at the Authorization Page. As for the Callback URIs: An application may have different addresses where they allow their users to login. Some may login from the application's homepage, some others may log in from other pages like their public profiles or something else. Most applications will want to have their users redirected back to the url they came from when they clicked "Login with Dock", we call those the "Callback URIs".
 
-As a result of registering, clients are given a ``client_id`` and a ``client_secret`` that they can use to sign their requests to our api.
+Step 2 - Authorization page
+---------------------------
 
-State Variable
---------------
-Clients are encouraged to use the `state` variable to prevent CSRF attacks. A good state variable could be a self-signed string containing some simple info like:
+The User is presented with a page in the Dock application where they need to:
 
-- Current url to redirect the user to the right page once back
-- User_id
+i) Authenticate (if they haven't already done so)
+ii) Check Application info (name, description, logo, etc)
```

Comment:
I'd change it to "ii) Check the Application's info (name, description, logo, etc)"